tests/unit/test_current_user.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+from mz_bokeh_package.utilities import CurrentUser
+from mz_bokeh_package.utilities import MZGraphQLClient
+
+API_KEY = "6fzQxEJL1YX9ldWUhHv5NPpiIcV7Gg8r03KwmeCqyRakj2nSAstDMTb4oB"
+SESSION_ID = "LME0EBfddYQCJqpZoO4ysYvynDXFLavC8hu3RzHW6e5e"
+USER_ID = "79e8e0f4-d2d6-4c16-b833-0a8986a6ce53"
+USER_NAME = "user_name"
+
+
+TESTS = [
+    # Test 1: session ID provided and is cached in CurrentUser._users. API key not required.
+    {
+        "input": {
+            "session_id": SESSION_ID,
+            "users_cache": {SESSION_ID: {"id": USER_ID, "name": USER_NAME}},
+            "api_key": None,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 2: session ID provided and is not cached in CurrentUser._users. API key provided.
+    {
+        "input": {
+            "session_id": SESSION_ID,
+            "users_cache": {},
+            "api_key": API_KEY,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 3: session ID not provided and is not cached in CurrentUser._users. API key provided.
+    {
+        "input": {
+            "session_id": None,
+            "users_cache": {},
+            "api_key": API_KEY,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+]
+
+
+@pytest.mark.parametrize("parameters", TESTS)
+def test_get_user_info(monkeypatch, parameters: dict):
+
+    session_id = parameters["input"]["session_id"]
+    monkeypatch.setattr(CurrentUser, "_get_session_id", lambda: session_id)
+
+    users_cache = parameters["input"]["users_cache"]
+    monkeypatch.setattr(CurrentUser, "_users", users_cache)
+
+    monkeypatch.setattr(MZGraphQLClient, "get_user", _get_user)
+
+    user_info = CurrentUser.get_user_info(parameters['input']['api_key'])
+    assert user_info == parameters['output']
+
+
+def test_get_user_info_error():
```

Comment:
TODO:
1. Add `TESTS_FAILURE` list with failing scenarios (think what the failing scenarios are).
2. Add fixture that prepares the input for both tests and use the fixture in both tests (to avoid code duplication).